### ISSUE TITLE
Cop to ensure the same style is used for indenting heredocs on a project

### ISSUE THREAD

**Issue by deivid-rodriguez** [rubocop-hq/rubocop#6512]
**Is your feature request related to a problem? Please describe.**

Rubocop can currently enforce nicely indented heredocs through the `Layout/IndentHeredoc` cop. It cannot, however, ensure that the same indentation style (`squiggly`, `unindent`, `strip_heredoc`, or `strip_indent`) is used for all heredocs on a project.

**Describe the solution you'd like**

I'm not sure whether this should be a different cop from `Layout/IndentHeredoc`. I tend to think it should be a different cop, although it would have the same enforced styles as `Layout/IndentHeredoc`.

**Comment by bquorning** [comment]
Maybe I misunderstand the issue, but don’t `Layout/IndentHeredoc`’s `EnforcedStyle` options “squiggly”, “active_support”, “powerpack”, and “unindent” solve exactly this problem?

**Comment by deivid-rodriguez** [comment]
Yeah, I had the same expectations, but it turns out this is not what `Layout/IndentHeredoc` tries to achieve. `Layout/IndentHeredoc` ensures that heredocs don't break the indentation flow of a source file. Its enforced styles are used for offense advices and auto-correction, but if a heredoc already has the proper indentation, even if it uses a style inconsistent with the rest of the project to achieve that, it won't trigger an offense.

**Comment by bquorning** [comment]
Ahh, I see. So with the configuration
```yaml
Layout/IndentHeredoc:
  EnforcedStyle: squiggly
```
this bit of Ruby should cause an offense
```rb
puts <<RUBY.strip_heredoc
  2 + 2
RUBY
```
which it currently doesn’t.

**Comment by deivid-rodriguez** [comment]
Yeah, either that, or have a different cop to enforce the consistency.
